### PR TITLE
[video_player_videohole] Implement 'httpHeaders' of 'VideoPlayerController.network'.

### PR DESCRIPTION
**Pull request by hyue7**
Close https://github.com/flutter-tizen/plugins/issues/621

### CODE REVIEW

File: packages/video_player_videohole/tizen/src/video_player.cc [hunk 1]
```diff
@@ -105,6 +106,34 @@ int64_t VideoPlayer::Create(const std::string &uri, int drm_type,
...
+        std::string user_agent = std::get<std::string>(iter->second);
+        ret = player_set_streaming_user_agent(player_, user_agent.c_str(),
+                                              user_agent.size());
+        if (ret != PLAYER_ERROR_NONE) {
+          LOG_ERROR("[MediaPlayer] player_set_streaming_user_agent failed: %s.",
+                    get_error_message(ret));
+        }
+      }
+    }
+  }
```
Comment (xiaowei-guan) [inline packages/video_player_videohole/tizen/src/video_player.cc:181]: suggest extract a new method to get value from http headers, for example:
```cpp
template <typename T>
void ExtractValueFromMap(const flutter::EncodableValue &arguments,
                         const char *key, T &outValue) {
  if (std::holds_alternative<flutter::EncodableMap>(arguments)) {
    flutter::EncodableMap values = std::get<flutter::EncodableMap>(arguments);
    flutter::EncodableValue value = values[flutter::EncodableValue(key)];
    if (std::holds_alternative<T>(value)) outValue = std::get<T>(value);
  }
}
```

File: packages/video_player_videohole/tizen/src/video_player_tizen_plugin.cc [hunk 2]
```diff
@@ -145,11 +146,17 @@ ErrorOr<PlayerMessage> VideoPlayerTizenPlugin::Create(
         }
       }
     }
+
+    const flutter::EncodableMap *http_headers_map = msg.http_headers();
+    if (http_headers_map) {
+      http_headers = *http_headers_map;
```
Comment (xiaowei-guan) [inline packages/video_player_videohole/tizen/src/video_player_tizen_plugin.cc:152]: Copying the http_header_map is not recommended.

File: packages/video_player_videohole/tizen/src/video_player.cc (hunk 1 above)
Comment (hyue7) [reply packages/video_player_videohole/tizen/src/video_player.cc]: fixed

File: packages/video_player_videohole/tizen/src/video_player_tizen_plugin.cc (hunk 2 above)
Comment (hyue7) [reply packages/video_player_videohole/tizen/src/video_player_tizen_plugin.cc]: fixed

**Review by xiaowei-guan — APPROVED** [review]

**Pull request merged**